modules/client-handler/src/integrationTest/java/org/apache/ignite/client/handler/ItSslClientHandlerTest.java

```diff
@@ -0,0 +1,144 @@
+        // Given server started
+        testServer = new TestServer();
+        serverModule = testServer.start(testInfo);
+
+        // Then
+        performAndCheckHandshake();
+    }
+
+    @Test
+    @DisplayName("When ssl configured on the server but not configured on the client then exception is thrown")
+    void sslConfiguredOnlyForServer(TestInfo testInfo) {
+        // Given server started
+        testServer = new TestServer(
+                TestSslConfig.builder()
+                        .keyStorePath(keyStorePkcs12Path)
+                        .keyStorePassword(password)
+                        .build()
+        );
+        serverModule = testServer.start(testInfo);
+
+        // Then
+        assertThrows(SocketException.class, this::performAndCheckHandshake);
+    }
+
+    private void performAndCheckHandshake() throws IOException {
+        serverPort = serverModule.localAddress().getPort();
+
+        try (var sock = new Socket("127.0.0.1", serverPort)) {
+            OutputStream out = sock.getOutputStream();
+
+            // Magic: IGNI
+            out.write(MAGIC);
+
+            // Handshake.
+            var packer = MessagePack.newDefaultBufferPacker();
+            packer.packInt(0);
+            packer.packInt(0);
+            packer.packInt(0);
+            packer.packInt(7); // Size.
+
+            packer.packInt(3); // Major.
+            packer.packInt(0); // Minor.
+            packer.packInt(0); // Patch.
+
+            packer.packInt(2); // Client type: general purpose.
+
+            packer.packBinaryHeader(0); // Features.
+            packer.packMapHeader(0); // Extensions.
+
+            out.write(packer.toByteArray());
+            out.flush();
+
+            // Read response.
+            var unpacker = MessagePack.newDefaultUnpacker(sock.getInputStream());
+            var magic = unpacker.readPayload(4);
+            unpacker.skipValue(3); // LE int zeros.
+            var len = unpacker.unpackInt();
+            var major = unpacker.unpackInt();
+            var minor = unpacker.unpackInt();
+            var patch = unpacker.unpackInt();
+            var success = unpacker.tryUnpackNil();
+            assertTrue(success);
+
+            var idleTimeout = unpacker.unpackLong();
+            var nodeId = unpacker.unpackString();
+            var nodeName = unpacker.unpackString();
+            unpacker.skipValue(); // Cluster id.
+
+            var featuresLen = unpacker.unpackBinaryHeader();
+            unpacker.skipValue(featuresLen);
+
+            var extensionsLen = unpacker.unpackMapHeader();
+            unpacker.skipValue(extensionsLen);
+
+            assertArrayEquals(MAGIC, magic);
+            assertEquals(44, len);
+            assertEquals(3, major);
+            assertEquals(0, minor);
+            assertEquals(0, patch);
+            assertEquals(0, idleTimeout);
+            assertEquals("id", nodeId);
+            assertEquals("consistent-id", nodeName);
```

Comment:
We expect this code to fail, I think most of it is unnecessary. Magic header should be enough to trigger disconnect due to SSL handshake error.

Reply:
Agreed, will get rid of most assertions.